package/MDAnalysis/lib/formats/libdcd.pyx

```diff
@@ -0,0 +1,414 @@
+                            &self.nsavc, &self.delta, &self.nfixed, &self.freeind,
+                            &self.fixedcoords, &self.reverse_endian,
+                            &self.charmm, &c_remarks, &len_remarks)
+        if ok != 0:
+            raise IOError("Reading DCD header failed: {}".format(DCD_ERRORS[ok]))
+
+        if c_remarks != NULL:
+            py_remarks = <bytes> c_remarks[:len_remarks]
+            free(c_remarks)
+        else:
+            py_remarks = ""
+
+        self.n_dims = 3 if not self.charmm & DCD_HAS_4DIMS else 4
+        self.n_frames = self._estimate_n_frames()
+        self.b_read_header = True
+
+        # make sure fixed atoms have been read
+        self.read()
+        self.seek(0)
+
+        if sys.version_info[0] < 3:
+            py_remarks = unicode(py_remarks, 'ascii', "ignore")
+            py_remarks = str(py_remarks.encode('ascii', 'ignore'))
+        else:
+            if isinstance(py_remarks, bytes):
+                py_remarks = py_remarks.decode('ascii', 'ignore')
+
+        py_remarks = "".join(s for s in py_remarks if s in string.printable)
+
+        return py_remarks
+
+    def _estimate_n_frames(self):
+        extrablocksize = 48 + 8 if self.charmm & DCD_HAS_EXTRA_BLOCK else 0
+        self.firstframesize = (self.n_atoms + 2) * self.n_dims * sizeof(float) + extrablocksize
+        self.framesize = ((self.n_atoms - self.nfixed + 2) * self.n_dims * sizeof(float) +
+                          extrablocksize)
+        filesize = path.getsize(self.fname)
+        # It's safe to use ftell, even though ftell returns a long, because the
+        # header size is < 4GB.
+        self.header_size = fio_ftell(self.fp)
+        # TODO: check that nframessize is larger then 0, the c-implementation
+        # used to do that.
+        nframessize = filesize - self.header_size - self.firstframesize
+        return nframessize / self.framesize + 1
+
+    @property
+    def periodic(self):
+          return bool((self.charmm & DCD_IS_CHARMM) and
+                      (self.charmm & DCD_HAS_EXTRA_BLOCK))
+
+
+    def read(self):
+        if self.reached_eof:
+            raise IOError('Reached last frame in DCD, seek to 0')
+        if not self.is_open:
+            raise IOError("No file open")
+        if self.mode != 'r':
+            raise IOError('File opened in mode: {}. Reading only allow '
+                               'in mode "r"'.format('self.mode'))
+
+        cdef np.ndarray xyz = np.empty((self.n_atoms, 3), dtype=FLOAT,
+                                       order='F')
+        cdef np.ndarray unitcell = np.empty(6, dtype=DOUBLE)
+
+        cdef FLOAT_T[::1] x = xyz[:, 0]
+        cdef FLOAT_T[::1] y = xyz[:, 1]
+        cdef FLOAT_T[::1] z = xyz[:, 2]
+
+        first_frame = self.current_frame == 0
+
+        cdef int lowerb = 0
+        cdef int upperb = self.n_atoms - 1
+
+        ok = read_dcdsubset(self.fp, self.n_atoms, lowerb, upperb,
```

Comment:
Replacing `read_dcdsubset` with `read_dcdstep` currently breaks the handling of unit cell information (the associated unit tests fail). That may partly explain why I went with the less intuitively-named function, but perhaps I should figure out why this discrepancy exists. 

Given your comment above about until cell info, there's probably something that's not quite right about unit cell handling in `DCDFile`.